source/extensions/access_loggers/grpc/grpc_access_log_utils.cc

```diff
@@ -271,7 +271,14 @@ void Utility::extractCommonAccessLogProperties(
   }
 
   if (stream_info.upstreamInfo().has_value()) {
+#if defined(__GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdangling-reference"
```

Comment:
Looks like we have quite a few of these. Is `-Wdangling-reference` finding false positives? Should we consider just disabling this warning in our build globally?

Reply:
Yes, gcc implementation is buggy, you can find reports in other projects. The pattern here is unusual but the same, so maybe better keep it since we know when the false positive happens.